sdk/src/main/java/com/akkaserverless/javasdk/valueentity/ValueEntityProvider.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.akkaserverless.javasdk.valueentity;
+
+import com.akkaserverless.javasdk.lowlevel.ValueEntityHandler;
+import com.google.protobuf.Descriptors;
+
+public interface ValueEntityProvider<T> {
+
+  Descriptors.ServiceDescriptor serviceDescriptor();
+
+  String entityTypeHint();
```

Comment:
I think we use just `entityType` in Akka Serverless, without `Hint`

Reply:
Yes, and we should stop doing so. This is the type hint and we should name it so. The reason it wasn't done before is that it was a field in the annotation and we would break it. 

With the new providers and codegen we can rename it without breaking anything.

Reply:
> This is the type hint and we should name it so

Can you explain why you think this is better? I don't understand what extra information the 'Hint' suffix gives.

Reply:
OK, I found https://github.com/lightbend/akkaserverless-framework/pull/670#issuecomment-848234377 - so the reasoning is that this string does not refer to a Java type, but is a string associated with the entity.

Reply:
> The reason it wasn't done before is that it was a field in the annotation and we would break it.
>
> With the new providers and codegen we can rename it without breaking anything.

We're still not changing the proto annotation name, though, right? I'm not super convinced adding 'Hint' is helpful here.

Reply:
I don't mind to change that on another iteration (when we change the annotation) . 

I do find it's helpful for us to align how we name things. The function of this bit of string is exactly the same as the type hint in Akka so we should name it the same for our own sanity.